src/goto-symex/renaming_level.cpp

```diff
@@ -46,8 +46,14 @@ operator()(ssa_exprt ssa_expr, const namespacet &ns, unsigned thread_nr) const
 renamedt<ssa_exprt, L1> symex_level1t::
 operator()(renamedt<ssa_exprt, L0> l0_expr) const
 {
-  if(!l0_expr.get().get_level_1().empty())
+  if(
+    !l0_expr.get().get_level_1().empty() ||
+    !l0_expr.get().get_level_2().empty())
+  {
+    // TODO reaching here means the information that l0_expr was L1 or L2 has
+    // TODO been lost somewhere during execution, which should be avoided
```

Comment:
Use `UNREACHABLE` (here and below)?

Reply:
Unfortunately it is reachable. There are even cases where level 2 is set but not level 1.

Reply:
Level 2 being set without level 1 is expected for objects with static lifetime, but maybe there are also actually buggy cases?

Reply:
That's good to know. I think it will be easier to see if something is done wrong once the renamedt type is used everywhere in goto_symex_state. (just to be clear, we do not change the behaviour here, the ignoring of the non-empty case was just done in the caller of this function).